### PR TITLE
Allow blank signing key for queries on Ethereum

### LINKED ISSUE #1485: FireFly requires a signing key for "query" operations
When using a `/query` API to query a smart contract, I'm required to supply a `key` or to have a default signing key configured on my namespace.

At least on Ethereum, I don't think this should be a requirement, since `eth_call` does not require any signing.

### PR DESCRIPTION
**Pull request by awrichar**
Fixes #1485

### CODE REVIEW

File: internal/blockchain/tezos/tezos.go [hunk 1]
```diff
@@ -283,6 +283,10 @@ func (t *Tezos) RemoveFireflySubscription(ctx context.Context, subID string) {
 }
 
 func (t *Tezos) ResolveSigningKey(ctx context.Context, key string, intent blockchain.ResolveKeyIntent) (resolved string, err error) {
+	// Key is always required
+	if key == "" {
```
Comment (awrichar) [inline internal/blockchain/tezos/tezos.go:287]: Could use confirmation from @denisandreenko on if this is correct.

File: internal/blockchain/ethereum/ethereum.go [hunk 2]
```diff
@@ -542,6 +542,13 @@ func formatEthAddress(ctx context.Context, key string) (string, error) {
 }
 
 func (e *Ethereum) ResolveSigningKey(ctx context.Context, key string, intent blockchain.ResolveKeyIntent) (resolved string, err error) {
+	// Key may be unset for query intent only
+	if key == "" {
+		if intent == blockchain.ResolveKeyIntentQuery {
+			return "", nil
```
Comment (awrichar) [inline internal/blockchain/ethereum/ethereum.go:548]: This is the primary needed change - everything else is just supporting this and/or general cleanup.

File: internal/blockchain/tezos/tezos.go (hunk 1 above)
Comment (awrichar) [reply internal/blockchain/tezos/tezos.go:287]: This is preserving existing behavior (the empty string check has been pushed from the identity manager down to the blockchain plugins) - but I'm happy to tweak if there are protocol-specific rules (such as how Ethereum allows you to query without a key).

File: internal/identity/identitymanager.go [hunk 3]
```diff
@@ -311,12 +311,11 @@ func (im *identityManager) getDefaultVerifier(ctx context.Context, intent blockc
...
 		orgKey := im.multiparty.RootOrg().Key
-		if orgKey == "" {
-			return nil, i18n.NewError(ctx, coremsgs.MsgNodeMissingBlockchainKey)
+		if orgKey != "" {
+			return im.resolveInputKeyViaBlockchainPlugin(ctx, orgKey, intent)
 		}
-		return im.resolveInputKeyViaBlockchainPlugin(ctx, orgKey, intent)
 	}
-	return nil, i18n.NewError(ctx, coremsgs.MsgNodeMissingBlockchainKey)
+	return im.resolveInputKeyViaBlockchainPlugin(ctx, "", intent)
```
Comment (awrichar) [inline internal/identity/identitymanager.go:318]: Allow the blockchain plugin to decide if `""` is allowed or not.

File: internal/blockchain/tezos/tezos.go (hunk 1 above)
Comment (alex-semenyuk) [reply internal/blockchain/tezos/tezos.go:287]: Yes, this is correct.

Actually it gives error later at formatTezosAddress if key is empty string, but MsgNodeMissingBlockchainKey looks more specific which is better I believe. 

Thanks for catching this.

Comment (denisandreenko) [reply internal/blockchain/tezos/tezos.go:287]: Looks good to me

**Review by denisandreenko — APPROVED** [review]

**Review by alex-semenyuk — APPROVED** [review]

**Review by EnriqueL8 — COMMENTED** [review]
Looks good to me - one small comment

File: internal/blockchain/ethereum/ethereum.go [hunk 4]
```diff
@@ -643,7 +650,7 @@ func (e *Ethereum) queryContractMethod(ctx context.Context, address, signingKey
 	return res, nil
 }
 
-func (e *Ethereum) buildBatchPinInput(ctx context.Context, version int, namespace string, batch *blockchain.BatchPin) (*abi.Entry, []interface{}) {
```
[the remark this answers is not in the document]
Comment (EnriqueL8) [reply internal/blockchain/ethereum/ethereum.go:646]: I guess it wasn't use to log anything?

[the remark this answers is not in the document]
Comment (awrichar) [reply internal/blockchain/ethereum/ethereum.go:646]: Yea, I think the new version of Go (or some integration in VS Code) flags additional linting problems, like unused variables. I went ahead and accepted a few of its suggestions in the files I was editing.

**Review by peterbroadhurst — APPROVED** [review]

**Pull request merged**